python_linear_regression/python_linear_regression.md

```diff
@@ -0,0 +1,453 @@
+<!--
+
+author:   Daniel Schwartz
+email:    des338@drexel.edu
+version:  0.0.0
+current_version_description: Initial version
+module_type: standard
+docs_version: 3.0.0
+language: en
+narrator: UK English Female
+mode: Textbook
+
+title: Python Lesson on Regression for Machine Learning
+
+comment:  Understand the basics of linear regression, including how it works and when to use it in Python.
+
+long_description: Understand linear regression in Python as a powerful and versatile tool that can be used to solve a wide variety of problems. By understanding the key concepts and techniques involved in linear regression, you can build and deploy models that can accurately predict the target variable of interest.
+
+estimated_time_in_minutes: 20
+
+@pre_reqs
+
+This module assumes learners have been exposed to introductory statistics, Algebra, and probability.
+There are coding exercises in Python, so programming experience is required.
+
+@end
+
+@learning_objectives  
+
+-   Understand the concept of linear regression and its applications in machine learning
+-   Learn how to implement the linear regression algorithm in Python
```

Comment:
These are both pretty big topics, actually, and I'm wondering now if it might be worthwhile splitting this module into two separate modules: "Intro to Linear Regression for Machine Learning", and "Linear Regression in Python". 
That will allow you to focus more attention on actually teaching the python code, which would be great. The prereqs don't list specific experience with scikit learn or anything like that, so we want to write this with a learner in mind who has some python experience but has maybe never done machine learning before. Linear regression is a really natural place to start with modeling, so I love the idea of this module being something someone could work through as their first attempt at machine learning in python.
I'll start a new branch for the intro to linear regression module, and we can keep this one for the python piece.